test/unit/core/main.js

```diff
@@ -31,6 +31,91 @@ suite('Core', function() {
         p5.prototype._registeredMethods.init = originalInit;
       }
     });
+    test('should register and call before and after "preload" hooks', function() {
+      return new Promise(resolve => {
+        let beforePreloadCalled = false;
+        let preloadCalled = false;
+        let afterPreloadCalled = false;
+
+        p5.prototype.registerMethod('beforePreload', () => {
+          beforePreloadCalled = true;
+        });
+
+        p5.prototype.registerMethod('preload', () => {
+          assert.equal(beforePreloadCalled, true);
+          preloadCalled = true;
+        });
+
+        p5.prototype.registerMethod('afterPreload', () => {
+          if(beforePreloadCalled && preloadCalled)
+            afterPreloadCalled = true;
+        });
+
+        myp5 = new p5(function(sketch) {
+          sketch.preload = () => {};
+          sketch.setup = () => {
+            assert.equal(afterPreloadCalled, true);
+          };
+          resolve();
```

Comment:
Currently I think this will resolve before setup or preload gets run. Does it work to move the `resolve` into `sketch.setup` after your assertion?